app/code/Magento/Cms/Model/Page/DataProvider.php

```diff
@@ -128,29 +122,54 @@ public function getData()
         if (isset($this->loadedData)) {
             return $this->loadedData;
         }
-        $this->collection = $this->collectionFactory->create();
-        $items = $this->collection->getItems();
-        /** @var $page \Magento\Cms\Model\Page */
-        foreach ($items as $page) {
-            $this->loadedData[$page->getId()] = $page->getData();
-            if ($page->getCustomLayoutUpdateXml() || $page->getLayoutUpdateXml()) {
-                //Deprecated layout update exists.
-                $this->loadedData[$page->getId()]['layout_update_selected'] = '_existing_';
+
+        $page = $this->getCurrentPage();
+        $this->loadedData[$page->getId()] = $page->getData();
+        if ($page->getCustomLayoutUpdateXml() || $page->getLayoutUpdateXml()) {
+            //Deprecated layout update exists.
+            $this->loadedData[$page->getId()]['layout_update_selected'] = '_existing_';
+        }
+
+        return $this->loadedData;
+    }
+
+    /**
+     * Return current page
+     *
+     * @return PageInterface
+     */
+    private function getCurrentPage(): PageInterface
+    {
+        $newPage = $this->collection->getNewEmptyItem();
```

Comment:
Let's use factory to create a new object instead, the `getNewEmptyItem` returns `DataObject` and this method must return `PageInterface`.

Also, I would avoid assigning a variable if it is not gonna be used, I rather call directly the create method on the factory whenever it is necessary.